test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,270 @@
+          it('defaults to stderr', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: '' },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions.logDestination).to.equal('stderr');
+          });
+        });
+
+        context('when MONGODB_LOG_PATH is stdout', function () {
+          it('treats the value as case-insensitive', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: 'STDOUT' },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions).to.have.property('logDestination', 'stdout');
+          });
+        });
+
+        context('when MONGODB_LOG_PATH is stderr', function () {
+          it('treats the value as case-insensitive', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: 'STDERR' },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions).to.have.property('logDestination', 'stderr');
+          });
+        });
+
+        context('when MONGODB_LOG_PATH is a non-empty string', function () {
+          it('uses the passed value', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: '/path/to/file.txt' },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions.logDestination).to.equal('/path/to/file.txt');
+          });
+        });
+      });
+
+      context('when MONGODB_LOG_PATH is undefined', function () {
+        context('when mongodbLogPath is an empty string', function () {
+          it('defaults to stderr', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: undefined },
+              { mongodbLogPath: '' }
+            );
+            expect(loggerOptions.logDestination).to.equal('stderr');
+          });
+        });
+
+        context('when mongodbLogPath is stdout', function () {
+          it('treats the value as case-insensitive', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: undefined },
+              { mongodbLogPath: 'STDOUT' }
+            );
+            expect(loggerOptions).to.have.property('logDestination', 'stdout');
+          });
+        });
+
+        context('when mongodbLogPath is stderr', function () {
+          it('treats the value as case-insensitive', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: undefined },
+              { mongodbLogPath: 'STDERR' }
+            );
+            expect(loggerOptions).to.have.property('logDestination', 'stderr');
+          });
+        });
+
+        context('when mongodbLogPath is a non-empty string', function () {
```

Comment:
Agreed about using an invalid non-empty string, will update the description and probably the test structure.  

We do test the empty string already:
```typescript
context('when mongodbLogPath is an empty string', function () {
  it('defaults to stderr', function () {
    const loggerOptions = MongoLogger.resolveOptions(
      { MONGODB_LOG_PATH: undefined },
      { mongodbLogPath: '' }
    );
    expect(loggerOptions.logDestination).to.equal('stderr');
  });
});
```

w.r.t. `null` or `0` or other weird values - this is a larger discussion.  

The TS type for `mongodbLogPath` is `string | undefined.  Other falsey values (`0`, `null`, ect) are prohibited by Typescript.  An argument could be made that since we also ship the compiled JS, we should test other values too.  But thus far, our pattern has been not to test for values outside of our Typescript and my opinion is that it is perfectly reasonable to only test edge cases within our types.  Our types define the api contract of our functions for both our js and typescript users - anyone providing an object that doesn't match our type definition is working outside the bounds of our supported API.

Reply:
I left a comment regarding this in a separate thread in my most recent set of comments for easier discussion